Core/src/ExponentialBackoff.php

```diff
@@ -79,6 +90,11 @@ public function execute(callable $function, array $arguments = [])
             try {
                 return call_user_func_array($function, $arguments);
             } catch (\Exception $exception) {
+                if ($this->onRetryException) {
```

Comment:
If we want, we can actually split this in 2 as well, a function that simply consumes the exception and another that is used to modify the headers.

Reply:
If we are doing this then one of the options can be to rename the above callback from `$onExecutionStart` to something like `$consumeArguments`, thus delegating the task of modifying `$request`/`$options` to this callback. Thus we would only need to create a callback for consume exception.

Reply:
The `on...` pattern signifies a callback/hook(wordpress terminology), so it becomes clear just by reading when the function will be executed.

Although we aren't really splitting the callbacks as of now.